test/tests/functional/pbs_alps_request_tunables.py

```diff
@@ -0,0 +1,150 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+import math
+
+
+@tags('cray')
+class TestCrayAlpsRequestTunables(TestFunctional):
+    """
+    Set of tests to verify alps release tunables namely alps_release_interval
+    and alps_release_jitter
+    """
+
+    def setUp(self):
+        machine = self.du.get_platform()
+        if not machine.startswith('cray') or machine == "craysim":
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    def get_epoch(self, msg):
+        a = time.strptime(msg[:19], "%m/%d/%Y %H:%M:%S")
+        return int(time.mktime(a))
+
+    def test_alps_release_interval(self):
+        """
+        Set alps_release_interval to a higher value and then notice that
+        subsequest reservation cancellation requests are made atleast
+        after the set interval.
+        """
+        ari = 4.298
+        self.mom.add_config({'$alps_release_interval': ari})
+
+        # submit a job and then delete it after it starts running
+        start_time = time.time()
+        j1 = Job(TEST_USER)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1)
+
+        # Look for a message that confirms that reservation is deleted
+        self.mom.log_match("%s;ALPS reservation cancelled" % jid1,
+                           starttime=start_time, max_attempts=5,
+                           interval=2)
+        # Now that we know that reservation is cleared we should
+        # check for time difference between each cancellation request
+        out = self.mom.log_match("%s;Canceling ALPS reservation *" % jid1,
+                                 n='ALL', max_attempts=5, regexp=True,
+                                 allmatch=True)
+
+        # We found something, Let's first check there are atleast 2 such
+        # log messages, If not then that means reservation was cancelled
+        # in the first attempt itself, at that point right thing to do is
+        # to either run it again or find out a way to delay the reservation
+        # cancellation at ALPS level itself.
+        if len(out) >= 2:
+            nline_prev, line_prev = out[0]
+            time_prev = self.get_epoch(line_prev)
```

Comment:
Looks like nline_prev is not used.  Also, if you only use line_prev once, just use out[0][1].